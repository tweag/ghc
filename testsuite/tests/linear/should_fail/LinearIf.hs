{-# LANGUAGE LinearTypes #-}
{-# LANGUAGE RebindableSyntax #-}

module LinearIf where

import Prelude (Bool(..), Char)

ifThenElse :: Bool -> a -> a -> a
ifThenElse True x _ = x
ifThenElse False _ y = y

f :: Bool #-> Char #-> Char #-> Char
f b x y = if b then x else y
  -- 'f' ought to be unrestricted in all three arguments because it desugars to
  -- > ifThenElse b x y
