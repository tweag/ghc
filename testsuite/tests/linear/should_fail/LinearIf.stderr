
LinearIf.hs:13:3: error:
    • Couldn't match type ‘'Many’ with ‘'One’
        arising from multiplicity of ‘b’
    • In an equation for ‘f’: f b x y = if b then x else y

LinearIf.hs:13:5: error:
    • Couldn't match type ‘'Many’ with ‘'One’
        arising from multiplicity of ‘x’
    • In an equation for ‘f’: f b x y = if b then x else y

LinearIf.hs:13:7: error:
    • Couldn't match type ‘'Many’ with ‘'One’
        arising from multiplicity of ‘y’
    • In an equation for ‘f’: f b x y = if b then x else y
